crates/resolver/src/tls/dns_over_rustls.rs

```diff
@@ -29,9 +29,19 @@ const ALPN_H2: &[u8] = b"h2";
 lazy_static! {
     // using the mozilla default root store
     pub(crate) static ref CLIENT_CONFIG: Arc<ClientConfig> = {
+        #[cfg_attr(not(any(feature = "native-certs", notfeature = "webpki-roots")), allow(unused_mut))]
         let mut root_store = RootCertStore::empty();
+        #[cfg(all(feature = "native-certs", not(feature = "webpki-roots")))]
+        root_store.add_parsable_certificates(
+            &rustls_native_certs::load_native_certs()
+                .unwrap()
+                .into_iter()
+                .map(|cert| cert.0)
+                .collect::<Vec<_>>(),
+        );
```

Comment:
This needs to be fixed overall anyway. But what should I do with the `unwrap()` here?

If we intend to bubble up an error, this could be done here by replacing `lazy_static` with `once_cell`, which would be a good improvement anyway. Then we could use `OnceCell` to try and initialize it, when it fails, we can bubble up the error.

We could also just store a `Result` here, which wouldn't allow for retrying in the lifetime of the application.

Reply:
I actually started on a PR to rip out all the uses of lazy_static to replace them with once_cell, so that sounds okay to me. Though ideally we shouldn't add a dependency without removing one (but maybe once_cell is already used by trust-dns or some of its dependencies)?

Reply:
> I actually started on a PR to rip out all the uses of lazy_static to replace them with once_cell [..]

I'm happy to continue on that PR or make one from scratch. Didn't check if `once_cell` is in the dependency tree, but I would be willing to completely replace it anyway.

Reply:
My branch is here: https://github.com/bluejekyll/trust-dns/compare/once-cell?expand=1. Warning: it's a fair bit of work! Would be great to get it done, though. (Maybe my branch is already outdated because more things can just be `const` instead of having to be once_cell, with the current MSRV.)

Reply:
Alright, I just realized that properly implementing `rustls-native-certs` won't work, we have to handle the error somehow or panic. Returning the error isn't an option, it touches too many APIs, so it would be a breaking change.

1. Wait until the next breaking change, I will update the PR then.
2. Merge as is, which panics.
3. Ignore errors; basically load no certificates if `rustls-native-certs` fails.

Reply:
Breaking change is not a problem, we're currently releasing alphas anyway. But not sure how involved/impactful yielding the error would be?